scan-command/src/main/java/io/ballerina/scan/utils/ScanUtils.java

```diff
@@ -550,4 +555,91 @@ private static Optional<String> loadRemoteJAR(Path targetDir, String fileName, U
             return Optional.empty();
         }
     }
+
+    /**
+     * Prints the {@link List<Rule>} of static analysis rules to the console.
+     *
+     * @param rules        the list of static analysis rules
+     * @param outputStream the print stream
+     */
+    public static void printRulesToConsole(List<Rule> rules, PrintStream outputStream) {
+        int maxRuleIDLength = RULE_ID_COLUMN.length();
+        int maxSeverityLength = RULE_SEVERITY_COLUMN.length();
+        int maxDescriptionLength = RULE_DESCRIPTION_COLUMN.length();
+
+        for (Rule rule : rules) {
+            maxRuleIDLength = Math.max(maxRuleIDLength, rule.id().length());
+            maxSeverityLength = Math.max(maxSeverityLength, rule.kind().toString().length());
+            maxDescriptionLength = Math.max(maxDescriptionLength, rule.description().length());
+        }
+
+        String format = "\t%-" + maxRuleIDLength + "s | %-" + maxSeverityLength + "s | %-" + maxDescriptionLength
+                + "s%n";
+
+        outputStream.printf(format, RULE_ID_COLUMN, RULE_SEVERITY_COLUMN, RULE_DESCRIPTION_COLUMN);
+        outputStream.printf("\t" + "-".repeat(maxRuleIDLength + 1) + "--" +
+                "-".repeat(maxSeverityLength + 1) + "--" + "-".repeat(maxDescriptionLength + 1) + "%n");
+
+        sortRules(rules);
+        for (Rule rule : rules) {
+            String formattedLine = String.format(format, rule.id(), rule.kind().toString(), rule.description());
+            outputStream.println(formattedLine.stripTrailing());
+        }
+    }
+
+    /**
+     * <p> Sorts the rules based on a specified order. </p>
+     * <p> The rules are sorted based on the following order. </p>
+     * <ul>
+     *  <li> -1 if the first element should come before the second element </li>
+     *  <li>  1 if the second element should come before the first element </li>
+     * </ul>
+     * <p> The sorting priority is provided as follows. </p>
+     * <ol>
+     *  <li> `ballerina` rules first </li>
+     *  <li> `ballerina/moduleName` rules next </li>
+     *  <li> `ballerinax/moduleName` rules next </li>
+     *  <li> `wso2/moduleName` rules next </li>
+     *  <li> All remaining rules sorted alphabetically </li>
+     * </ol>
+     *
+     * @param rules The list of rules to be sorted.
+     */
+    private static void sortRules(List<Rule> rules) {
+        Comparator<Rule> ruleComparator = (r1, r2) -> {
+            String id1 = r1.id().split(":")[0];
+            String id2 = r2.id().split(":")[0];
+
+            if (id1.equals("ballerina") && !id2.equals("ballerina")) {
```

Comment:
Why do we have both "ballerina" and "ballerina/" here? Can't we strip `/` when creating id1, id2 and remove the "ballerina/" part?

Reply:
There are two types of rules used by the scan tool,

- Core rules
- External rules

The fully qualified rule names of core rules follows the format `ballerina:1` while external rules follow the format `org/name:1`

So for example if the ballerina io module introduces static analysis rules via a compiler plugin, there fully qualified rule identifier would look like `ballerina/io:1`, and  if there are rules introduced by the ballerina http module it would be as `ballerina/http:1`

Also, we should give priority to sorting in the order:
1. Core rules (rules built into the scan tool)
2. ballerina org compiler plugin rules
3. ballerinax org compiler plugin rules
4. wso2 org compiler plugin rules
5. external org compiler plugin rules.

So if we also have rules introduced by an external compiler plugin where the org is arc and the name is http, the printed output should be ordered in the following manner:
```bash
ballerina:1
ballerina/http:1
ballerina/io:1
arc/http:1
```

if we remove the `ballerina/` part, then This order cannot be preserved.